pkg/build/build.go

```diff
@@ -16,10 +21,43 @@ const (
 )
 
 type Build struct {
-	Id      string          `json:"id" validate:"required"`
-	Hash    string          `json:"hash" validate:"required"`
-	State   BuildState      `json:"state" validate:"required"`
-	Project project.Project `json:"project" validate:"required"`
-	User    string          `json:"user" validate:"required"`
-	Image   string          `json:"image" validate:"required"`
+	Id          string                     `json:"id" validate:"required"`
+	Hash        string                     `json:"hash" validate:"required"`
+	State       BuildState                 `json:"state" validate:"required"`
+	Image       string                     `json:"image" validate:"required"`
+	User        string                     `json:"user" validate:"required"`
+	BuildConfig *buildconfig.BuildConfig   `json:"buildConfig" validate:"optional"`
+	Repository  *gitprovider.GitRepository `json:"repository" validate:"optional"`
+	EnvVars     map[string]string          `json:"envVars" validate:"required"`
+	PrebuildId  string                     `json:"prebuildId" validate:"required"`
+	CreatedAt   time.Time                  `json:"createdAt" validate:"required"`
+	UpdatedAt   time.Time                  `json:"updatedAt" validate:"required"`
 } // @name Build
+
+func FetchCachedBuild(build *Build, builds []*Build) *buildconfig.CachedBuild {
```

Comment:
I suggest we use `Get` or `Find` instead of `Fetch`. `Fetch` implies a request of some kind